datalake-api/src/main/java/com/sequenceiq/sdx/api/model/SdxClusterRequestBase.java

```diff
@@ -0,0 +1,112 @@
+package com.sequenceiq.sdx.api.model;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.validation.Valid;
+import javax.validation.constraints.NotNull;
+
+import com.sequenceiq.common.api.tag.request.TaggableRequest;
+
+public class SdxClusterRequestBase implements TaggableRequest {
+
+    @NotNull
+    private String environment;
+
+    @NotNull
+    private SdxClusterShape clusterShape;
+
+    private SdxCloudStorageRequest cloudStorage;
+
+    private SdxDatabaseRequest externalDatabase;
+
+    @Valid
+    private SdxAwsRequest aws;
```

Comment:
I know this is not part of the PR, but perhaps it made sense to add a proper `toString()` implementation for the classes involved here. `SdxClusterShape` is an enum, so it will work as expected, but the other three types will show the default (and pretty useless) `Object.toString()` output.

Reply:
I have created CB-11443 to track that effort. So I can focus on the feature in this PR.